internal/heartbeat/heartbeat.go

```diff
@@ -121,3 +121,11 @@ func (s *Heartbeat) Update(config models.DeviceConfigurationMessage) error {
 	s.initTicker(periodSeconds)
 	return nil
 }
+
+func (s *Heartbeat) Deregister() error {
+	log.Info("Stopping heartbeat ticker")
```

Comment:
Can you add to initTicker a message when the loop is exited that the heartbeat is stopped ?